evm_loader/lib/src/abi/mod.rs

```diff
@@ -0,0 +1,195 @@
+mod cancel_trx;
+mod collect_treasury;
+mod emulate;
+mod get_balance;
+mod get_config;
+mod get_contract;
+mod get_holder;
+mod get_neon_elf;
+mod get_storage_at;
+mod init_environment;
+mod trace;
+
+use crate::{
+    config::{self},
+    rpc::{CallDbClient, RpcEnum},
+    types::{RequestWithSlot, TracerDb},
+    Config, LibMethod, NeonError,
+};
+use abi_stable::{
+    prefix_type::WithMetadata,
+    sabi_extern_fn,
+    std_types::{RStr, RString},
+};
+use async_ffi::FutureExt;
+use clap::ArgMatches;
+use neon_lib_interface::{
+    types::{NeonEVMLibError, RNeonEVMLibResult},
+    NeonEVMLib,
+};
+use serde_json::json;
+use solana_clap_utils::keypair::signer_from_path;
+use solana_sdk::signer::Signer;
+
+pub const _MODULE_WM_: &WithMetadata<NeonEVMLib> = &WithMetadata::new(NeonEVMLib {
+    hash,
+    get_version,
+    get_build_info,
+    invoke,
+});
+
+#[sabi_extern_fn]
+fn hash() -> RString {
+    env!("NEON_REVISION").into()
+}
+
+#[sabi_extern_fn]
+fn get_version() -> RString {
+    env!("CARGO_PKG_VERSION").into()
+}
+
+#[sabi_extern_fn]
+fn get_build_info() -> RString {
+    json!(crate::build_info::get_build_info())
+        .to_string()
+        .into()
+}
+
+#[sabi_extern_fn]
+fn invoke<'a>(method: RStr<'a>, params: RStr<'a>) -> RNeonEVMLibResult<'a> {
+    async move {
+        dispatch(method.as_str(), params.as_str())
+            .await
+            .map(RString::from)
+            .map_err(neon_error_to_rstring)
+            .into()
+    }
+    .into_local_ffi()
+}
+
+async fn load_config() -> Result<Config, NeonError> {
+    let api_options = config::load_api_config_from_enviroment();
+    let config = config::create_from_api_config(&api_options)?;
+
+    Ok(config)
+}
+
+async fn dispatch(method_str: &str, params_str: &str) -> Result<String, NeonError> {
+    let method: LibMethod = method_str.parse()?;
+    let config = load_config().await?;
+    let RequestWithSlot {
+        slot,
+        tx_index_in_block,
+    } = match params_str {
+        "" => RequestWithSlot {
+            slot: None,
+            tx_index_in_block: None,
+        },
+        _ => serde_json::from_str(params_str).map_err(|_| params_to_neon_error(params_str))?,
+    };
+    let rpc = build_rpc(&config, slot, tx_index_in_block).await?;
```

Comment:
Currently, `let rpc = build_rpc(&config, slot, tx_index_in_block).await?;` creates either a new Solana or ClickHouse client for each library call, which is not efficient.

It should re-use the existing Solana or ClickHouse client, just like I did for neon-api in #194.

So I suggest moving the `neon_api::api_server::state` module from `neon-api` crate to `neon-lib` crate. Then a static `State` can be defined using `State::new` and `State::build_rpc` can be re-used just like in `neon-api` crate:

```Rust
lazy_static! {
    static ref RUNTIME: tokio::runtime::Runtime = tokio::runtime::Runtime::new().unwrap();
    static ref STATE: State = State::new(load_config().unwrap());
}
```

```Rust
    let state = &STATE;
    let rpc = state.build_rpc(slot, tx_index_in_block).await?;
    let config = &state.config;
```